src/helpers.js

```diff
@@ -118,19 +118,22 @@ function dotPathToHash(entry, target = {}, options = {}) {
  * Takes a `source` hash and makes sure its value
  * is pasted in the `target` hash, if the target
  * hash has the corresponding key (or if `options.keepRemoved` is true).
- * @returns An `{ old, new, mergeCount, pullCount, oldCount }` object.
+ * @returns An `{ old, new, mergeCount, pullCount, oldCount, reset }` object.
  * `old` is a hash of values that have not been merged into `target`.
  * `new` is `target`. `mergeCount` is the number of keys merged into
  * `new`, `pullCount` is the number of context and plural keys added to
  * `new` and `oldCount` is the number of keys that were either added to `old` or
  * `new` (if `options.keepRemoved` is true and `target` didn't have the corresponding
- * key).
+ * key) and `reset` is the keys that were reset due to not matching default values,
+ *  and `resetCount` which is the number of keys reset.
  */
-function mergeHashes(source, target, options = {}) {
+function mergeHashes(source, target, options = {}, resetValues = {}) {
```

Comment:
Maybe it's worth putting `resetValues` inside `options` since it was already the last optional argument?